org.eclipse.scanning.event.queues/src/org/eclipse/scanning/event/queues/QueueService.java

```diff
@@ -81,24 +83,42 @@
 	
 	/**
 	 * No argument constructor for OSGi
+	 * @throws URISyntaxException 
 	 */
 	public QueueService() {
-		
+		this(getQueueRootFromProperty(), CommandConstants.getScanningBrokerUri());
+	}
+	
+	private static final String getQueueRootFromProperty() {
+		String root = System.getProperty("org.eclipse.scanning.event.queues.queue.root");
+		if (root==null) root = System.getProperty("GDA/gda.event.queues.queue.root");
+		return root;
 	}
 	
+	public QueueService(String queueRoot, URI uri) {
```

Comment:
I *think* this is for testing, so it want's a comment above to constructor to that effect...?